### PR TITLE
Stop using Enum.IsDefined

### PR DESCRIPTION
**Pull request by ajcvickers**
It was taking 25% of the time in the Update perf test.

### CODE REVIEW

File: src/EntityFramework.Core/ChangeTracking/EntityEntry.cs [hunk 1]
```diff
@@ -58,7 +60,11 @@ public virtual EntityState State
             get { return _internalEntityEntry.EntityState; }
             set
             {
-                Check.IsDefined(value, nameof(value));
+                if (value < 0
+                    || (int)value > 4)
```
Comment (AndriySvyryd) [inline src/EntityFramework.Core/ChangeTracking/EntityEntry.cs]: Why magic numbers?

File: src/EntityFramework.Relational/Infrastructure/RelationalDbContextOptionsBuilder.cs [hunk 2]
```diff
@@ -52,7 +53,12 @@ public virtual TBuilder UseRelationalNulls()
 
         public virtual TBuilder QueryClientEvaluationBehavior(QueryClientEvaluationBehavior queryClientEvaluationBehavior)
         {
-            Check.IsDefined(queryClientEvaluationBehavior, nameof(queryClientEvaluationBehavior));
+            if (queryClientEvaluationBehavior < 0
+                || (int)queryClientEvaluationBehavior > 2)
```
Comment (AndriySvyryd) [inline src/EntityFramework.Relational/Infrastructure/RelationalDbContextOptionsBuilder.cs]: Magic numbers

File: src/EntityFramework.Core/ChangeTracking/EntityEntry.cs (hunk 1 above)
Comment (ajcvickers) [inline src/EntityFramework.Core/ChangeTracking/EntityEntry.cs]: Would you prefer named constants? Or using values from the enum itself? Or explicitly listing each value? None of these options will prevent this code from breaking if a new enum value is added, so I went for the simplest thing. Other ideas welcome, as long as the code is fast.

Comment (AndriySvyryd) [inline src/EntityFramework.Core/ChangeTracking/EntityEntry.cs]: Put the result of `Enum.GetValues(typeof(EntityState)).Cast<int>().Max()` in a static readonly field

Comment (ajcvickers) [inline src/EntityFramework.Core/ChangeTracking/EntityEntry.cs]: Will do.

**Comment by AndriySvyryd** [comment]
:shipit:

**Comment by ajcvickers** [comment]
Merged

**Pull request closed without merging**